### DISCUSSION TITLE
Generate Flashcards from Vault instead from File

### DISCUSSION THREAD

**Discussion by Flummic** [reuseman/flashcards-obsidian#123] [Ideas]
First of all thank you for this awesome plugin! I really enjoy using it.
Currently a limitation of the plugin is that it only generates Flashcards for the currently active file.

Ideally I'd love it if the "Generate Flashcards Command" would scan my entire vault (or specified) folders for new flashcards/changes and not only the currently active file!

**Comment by mounta11n** [comment]
I would also very much appreciate this feature. For me, my deck of medicine questions contains almost 4000 flashcards, divided into seven sub-decks. This makes no sense to me in this context, linking Anki to Obsidian and searching for my flashcards in a document somewhere among hundreds of other flashcards, let alone adding more notes, thoughts, etc. there. I would need each flashcard as a single Markdown file that I could isolate, focus on, view and edit at my leisure if I wanted to.

I wasn't aware of the fact that it ONLY works if the flashcards are created 1. from a document AND also have to 2. remain there (having already been given their own ID). Because I first thought I could create the flashcards and then just use refactor to split the document into multiple files. But when I then tap on the source in Anki, then the card is not found, because it is not searched for the ID, but for the name of the original file.... Unfortunately this behavior of the addon leads one of its purposes ad absurdum.
Also considering that you can't get the benefits of visual links in the graph view anymore when hundreds or thousands of links point to a single document.

I would be very happy if you could at least implement this feature that the path from Anki back to Obsidian initiates a search of the source by ID and not by filename.
